lib/tmdb_handler.rb

```diff
@@ -81,52 +82,72 @@ def tmdb_handler_add_movie(tmdb_id)
     popularity: @movie.popularity, runtime: @movie.runtime, mpaa_rating: @movie.mpaa_rating)
   end
 
-  def tmdb_handler_update_movie(tmdb_id)
-    @movie_url = "https://api.themoviedb.org/3/movie/#{tmdb_id}?api_key=#{ENV['tmdb_api_key']}&append_to_response=trailers,credits,similar,releases"
-    @result = JSON.parse(open(@movie_url).read, symbolize_names: true)
-    @movie_to_update = Movie.find_by(tmdb_id: tmdb_id)
-    @movie = MovieMore.tmdb_info(@result)
-
-    @movie_to_update.update_attributes(title: @movie.title, imdb_id: @movie.imdb_id, genres: @movie.genres,
-      actors: @movie.actors, adult: @result[:adult], backdrop_path: @movie.backdrop_path,
-      poster_path: @movie.poster_path, release_date: @movie.release_date, overview: @movie.overview,
-      trailer: @movie.trailer, director: @movie.director, director_id: @movie.director_id,
-      vote_average: @movie.vote_average, popularity: @movie.popularity, runtime: @movie.runtime,
-      mpaa_rating: @movie.mpaa_rating)
+  def self.tmdb_handler_update_movie(tmdb_id)
+    movie = Movie.find_by(tmdb_id: tmdb_id)
+    Raven.capture_message("movie not found in our db: #{tmdb_id}") && return unless movie
+
+    movie_url = "#{BASE_URL}/movie/#{tmdb_id}?api_key=#{ENV['tmdb_api_key']}&append_to_response=trailers,credits,similar,releases"
+    api_result = HTTParty.get(movie_url).deep_symbolize_keys rescue nil
+    Raven.capture_message("API request failed for tmdb_id: #{tmdb_id}") && return unless api_result && api_result[:id] == tmdb_id
```

Comment:
If the request to `HTTParty.get(movie_url)` doesn't find a movie, the result looks like:

`{:status_code=>34, :status_message=>"The resource you requested could not be found."}`

I added the rescue just to be safe in case anything unexpected breaks. 

Otherwise, if the hash with the status_message is returned, the `id` won't match, and we'll get notified via Sentry, and the rake task will stop iterating.